File: ruby/4.rb
class Integer
	def palindrome?
		str = self.to_s
		rev = str.reverse
		rev == str
	end
end

def solve
	product = 0
	for i in 100..999 do
		for j in 100..999	do
			candidate = i * j
			if (candidate > product && candidate.palindrome?)
				product = candidate
			end
		end
	end
	puts product
end

File: clojure/4.clj
(defn palindrome? [n]
	(let [rev (reverse (str n))])
	(= rev (reverse rev))) 


(defn solve
	(loop [x 100]
		(loop [y 100]
			if()))

;(range 100 1000)

(println (palindrome? 201))
